### PR TITLE
Add basic playbar support for podcasts

### PR DESCRIPTION
**Pull request by sputnick1124**
Small-ish PR to get basic podcast support kickstarted. I plan on adding more support for searching and following podcasts, but figured I'd tackle the low hanging fruit first.

Things added:
* playback progress bar
* seek forward/backward
* copy episode/show uri

### CODE REVIEW

File: rustfmt.toml [hunk 1]
```diff
@@ -1 +1,2 @@
 tab_spaces=2
+edition = "2018"
```
Comment (sputnick1124) [inline rustfmt.toml:2]: Not sure if there is another way to allow `rust-fmt` deal with 2018 (read: `async` code), but this shouldn't hurt anything, right?

**Comment by sputnick1124** [comment]
@Rigellute Any update on getting this approved? I'm making good headway on full podcast support over on https://github.com/sputnick1124/spotify-tui/tree/podcast-support-search and I'm almost ready to submit a PR for it, but it builds off of these commits, so it should definitely wait until this goes through.

**Review by Rigellute — APPROVED** [review]
So sorry for the long delay @sputnick1124! 

Your changes look great, as always. Thank you 🙏

File: rustfmt.toml (hunk 1 above)
Comment (sputnick1124) [reply rustfmt.toml:2]: Oh ok, that makes sense. I wonder what I might have done to my machine that made it complain for me...

**Comment by sputnick1124** [comment]
No worries! I didn't mean to be pushy! Thanks again for starting and maintaining this whole thing!

**Pull request merged**